poolqc/pooling_qc_sample.py

```diff
@@ -144,98 +134,67 @@ def update_title(lane_value, run_value):
     run['rRNA Contamination (%reads aligned)'] = run['rRNA Contamination (%reads aligned)']
 
     run = run.round(2)
     run = run.filter(
-        ['library', 'Run', 'LaneNumber', 'Index1', 'Index2', 'YieldQ30', '% Mapped to Coding', '% Mapped to Intronic',
-         '% Mapped to Intergenic',
-         'rRNA Contamination (%reads aligned)', 'Proportion Correct Strand Reads', 'Coverage'])
-
+        ['library', 'Run', 'LaneNumber', 'Index1', 'Index2', 'SampleNumberReads', '% Mapped to Coding',
+         '% Mapped to Intronic',
+         '% Mapped to Intergenic', 'rRNA Contamination (%reads aligned)', 'Proportion Correct Strand Reads',
+         'Coverage'])
+    run = run.sort_values('library')
     csv = run.to_csv(index=False)
     csv = 'data:text/csv;charset=utf-8,' + urllib.parse.quote(csv)
     run = run.drop(columns=['Run', 'LaneNumber'])
 
+    # Append Pass/Fail and Thresholds
+
     columns = [{'name': i, 'id': i,
                 'type': 'numeric'} for i in run.columns]
 
     data = run.to_dict('records')
 
+    # highlighting datatable cells/columns/rows
     style_data_conditional = [{
         'if': {'column_id': 'library'},
         'backgroundColor': 'rgb(222, 222, 222)'
-    },
-        {'if': {'column_id': '% Mapped to Coding',
-                'filter': '0 < {% Mapped to Coding} < 20'},
-         'backgroundColor': 'rgb(219, 75, 75)'},
+    }, {'if': {'column_id': '% Mapped to Coding',
+               'filter_query': '0 < {% Mapped to Coding} and {% Mapped to Coding} < 20'},
+        'backgroundColor': 'rgb(219, 75, 75)'},
         {'if': {'column_id': '% Mapped to Intronic',
-                'filter': '0 < {% Mapped to Coding} < 20'},
+                'filter_query': '0 < {% Mapped to Intronic} and {% Mapped to Intronic} < 20'},
          'backgroundColor': 'rgb(219, 75, 75)'},
         {'if': {'column_id': '% Mapped to Intergenic',
-                'filter': '0 < {% Mapped to Intergenic} < 15'},
+                'filter_query': '0 < {% Mapped to Intergenic} and {% Mapped to Intergenic} < 15'},
+         'backgroundColor': 'rgb(219, 75, 75)'},
+        {'if': {'column_id': 'rRNA Contamination (%reads aligned)',
+                'filter_query': '0 < {rRNA Contamination (%reads aligned)} and {rRNA Contamination (%reads aligned)} < 15'},
          'backgroundColor': 'rgb(219, 75, 75)'},
     ]
-    downloadtimedate = datetime.today().strftime('%Y-%m-%d')
-    download = 'PoolQC_%s_%s_%s.csv' % (downloadtimedate, run_alias, lane_alias)
-
-    return columns, data, style_data_conditional, csv, download
+    return columns, data, style_data_conditional, csv
 
 
-def update_sampleindices(run, rows, derived_virtual_selected_rows, colors):
+def update_sampleindices(run):
+    run = run.sort_values('library')
+    num_libraries = len(run['library'])
+    samples_passing_clusters = '%s/%s' % (sum(i > 20000 for i in run['SampleNumberReads']), num_libraries)
+    index_threshold = 50000000
```

Comment:
Put constants at the top of the file (outside a function), all-cap the variable name, and comment what they are.

Reply:
This value will eventually be a variable, not a constant, since it will be something the client enters or will be calculated depending on the run itself. For now, this value is just a placeholder

Reply:
Even though it's a placeholder for now, could you still move it to the top and comment it? This will hopefully make it easier to find later when we want to extract it or make it more dynamic.